server/sonar-process-monitor/src/test/java/org/sonar/application/process/StopRequestWatcherImplTest.java

```diff
@@ -0,0 +1,114 @@
+ */
+package org.sonar.application.process;
+
+import java.io.IOException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.sonar.application.FileSystem;
+import org.sonar.application.Scheduler;
+import org.sonar.application.config.AppSettings;
+import org.sonar.application.process.StopRequestWatcherImpl;
+import org.sonar.process.ProcessCommands;
+import org.sonar.process.ProcessProperties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+public class StopRequestWatcherImplTest {
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+  @Rule
+  public TestRule safeGuard = new DisableOnDebug(Timeout.seconds(10));
+
+  private AppSettings settings = mock(AppSettings.class, RETURNS_DEEP_STUBS);
+  private ProcessCommands commands = mock(ProcessCommands.class);
+  private Scheduler scheduler = mock(Scheduler.class);
+
+  @Test
+  public void do_not_watch_command_if_disabled() throws IOException {
+    enableSetting(false);
+    StopRequestWatcherImpl underTest = new StopRequestWatcherImpl(settings, scheduler, commands);
+
+    underTest.startWatching();
+    assertThat(underTest.isAlive()).isFalse();
+
+    underTest.stopWatching();
+    verifyZeroInteractions(commands, scheduler);
+  }
+
+  @Test
+  public void watch_stop_command_if_enabled() throws Exception {
+    enableSetting(true);
+    StopRequestWatcherImpl underTest = new StopRequestWatcherImpl(settings, scheduler, commands);
+    underTest.setDelayMs(1L);
+
+    underTest.startWatching();
+    assertThat(underTest.isAlive()).isTrue();
+    verify(scheduler, never()).terminate();
+
+    when(commands.askedForStop()).thenReturn(true);
+    verify(scheduler, timeout(1_000L)).terminate();
+
+    underTest.stopWatching();
+    assertThat(underTest.isAlive()).isFalse();
+  }
+
+  @Test
+  public void create_instance_with_default_delay() throws IOException {
+    FileSystem fs = mock(FileSystem.class);
+    when(fs.getTempDir()).thenReturn(temp.newFolder());
+
+    StopRequestWatcherImpl underTest = StopRequestWatcherImpl.create(settings, scheduler, fs);
+
+    assertThat(underTest.getDelayMs()).isEqualTo(500L);
+  }
+
+  @Test
+  public void stop_watching_commands_if_thread_is_interrupted() throws Exception {
+    enableSetting(true);
+    StopRequestWatcherImpl underTest = new StopRequestWatcherImpl(settings, scheduler, commands);
+
+    underTest.startWatching();
+    underTest.interrupt();
+
+    while (underTest.isAlive()) {
+      Thread.sleep(1L);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2925)